### PR TITLE
Functional tests for explorer GUI

### LINKED ISSUE #38: Functional tests for `apps.explorer`
We will implement the functional tests for the explorer GUI.

### PR DESCRIPTION
**Pull request by BECATRUE**
This will close #38.

I implemented the functional tests, but it is the almost same as the removed parts in #35.

### CODE REVIEW

File: .github/workflows/pylint.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ jobs:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
       run: |
+        sudo apt-get update
```
Comment (kangz12345) [inline .github/workflows/pylint.yml:22]: Why is this necessary?

File: tests/test_explorer.py [hunk 2]
```diff
@@ -29,19 +35,35 @@ def test_init_thread(self):
...
         parent = QObject()
         with mock.patch("iquip.apps.explorer._FileFinderThread.fetched") as mockedFetched:
             thread = explorer._FileFinderThread(path="path", widget=widget,
                                                 callback=mock.MagicMock(), parent=parent)
             thread.run()
             thread.wait()
-            mocked_run_command.assert_called_once_with("artiq_client ls path")
+            self.mocked_get.assert_called_once_with("http://127.0.0.1:8000/ls/",
+                                               params={"directory": "path"},
+                                               timeout=10)
```
Comment (kangz12345) [inline tests/test_explorer.py]: Please follow one of the following styles:
```python
object.method(arg1,
              arg2,
              arg3)
```
or
```python
object.method(
    arg1,
    arg2,
    arg3,
)
```
or
```python
object.method(
    arg1, arg2, arg3)
```

**Review by kangz12345 — CHANGES_REQUESTED** [review]

**Comment by kangz12345** [comment]
Please finish this PR first to release 0.1.0-alpha!

File: .github/workflows/pylint.yml (hunk 1 above)
Comment (BECATRUE) [reply .github/workflows/pylint.yml:22]: GitHub tests often failed during installing external modules.
So I applied the suggested solution in [test](https://github.com/snu-quiqcl/iquip/actions/runs/5330613285/jobs/9657522710).

File: tests/test_explorer.py (hunk 2 above)
Comment (BECATRUE) [reply tests/test_explorer.py]: Oh, sorry😅 I intended the first suggestion, but it seems to have been pushed back

**Review by kangz12345 — APPROVED** [review]
LGTM
